cilium: svc label exposure (#34063)
cilium-team/cilium: Documentation/network/kubernetes/kubeproxy-free.rst

| Local    | Local    | Node-local only         | Node-local only       |
+----------+----------+-------------------------+-----------------------+

Selective Service Exposure
**************************

By default, Kubernetes services are exposed on all nodes in the cluster. Sometimes it can be
desirable to expose services only on a subset of the nodes in the cluster. Cilium provides

> a special label for implementing this. On the node level, nodes can be labeled as follows:

Reviewer: Suggestion for clarity: more straightforward, mention the label name (to avoid readers believing that the `=beefy` part is relevant as well).

```suggestion
By default, Cilium exposes Kubernetes services on all nodes in the cluster. To expose a
service only on a subset of the nodes instead, use the ``service.cilium.io/node`` label for
the relevant nodes. For example, label a node as follows:
```

.. parsed-literal::

>   kubectl label node foo service.cilium.io/node=beefy

Reviewer: Prompt symbol optional, I prefer using one but that's your call. Let's use a `code-block`, though:

```suggestion
.. code-block:: shell-session

  $ kubectl label node node_name service.cilium.io/node=beefy
```

When adding a new service which should only be exposed to nodes with label ``service.cilium.io/node=beefy``

> the service can be installed as follows:

Reviewer: ```suggestion
To add a new service that should only be exposed to nodes with label ``service.cilium.io/node=beefy``, install the service as follows:
```

> .. parsed-literal::
> 
>   apiVersion: v1

Reviewer: ```suggestion
.. code-block:: yaml

  apiVersion: v1
```

  kind: Service
  metadata:
    name: example-service
    labels:
      service.cilium.io/node: beefy
  spec:
    selector:
      app: example
    ports:
      - port: 8765
        targetPort: 9376
    type: LoadBalancer

> Note that changing a node label after a service has been exposed matching that label does
> not automatically reconcile that service (e.g. remove upon mismatch). Thus an agent restart
> would be required for now. Generally it is advised to fixate the node label upon joining
> the Kubernetes cluster and retain it throughout the node's lifetime.

Reviewer: ```suggestion
Note that changing a node label after a service has been exposed matching that label does not
automatically update the list of nodes where the service is exposed. To update exposure of the
service after changing node labels, restart the Cilium agent. Generally it is advised to fixate the
node label upon joining the Kubernetes cluster and retain it throughout the node's lifetime.
```
